src/cascadia/TerminalCore/terminalrenderdata.cpp

```diff
@@ -59,8 +59,8 @@ const COLORREF Terminal::GetBackgroundColor(const TextAttribute& attr) const noe
 {
     const auto bgColor = attr.CalculateRgbBackground({ _colorTable.data(), _colorTable.size() }, _defaultFg, _defaultBg);
     // We only care about alpha for the default BG (which enables acrylic)
-    // If the bg isn't the default bg color, then make it fully opaque.
-    if (!attr.BackgroundIsDefault())
+    // If the bg isn't the default bg color, or reverse video is enabled, make it fully opaque.
+    if (!attr.BackgroundIsDefault() || WI_IsFlagSet(attr.GetMetaAttributes(), COMMON_LVB_REVERSE_VIDEO))

```

Comment:
I may file a followup TODO: the _foreground_ should become transparent in this case, which would just be wild. Also, it simply wouldn't work, because we would need to push it on as a clipping layer when we draw the background ;P

Reply:
On a related note, we may also want to consider whether a background color that simply matches the default background should also be made transparent. Right now, if I set my background to `\e[40m` or `\e[48;5;0m` those colors will also be treated as transparent. Once #2661 is fixed, though, that wouldn't be the case - only an explicit default background would be transparent. Maybe that's the way it should be - I don't what most people expect from this feature. We just need to be aware that the current behaviour _is_ going to change after #2661 if we leave things as they are.

Reply:
I think it's the _right_ thing to do (separate 40/48 and 37/38), but we may need to make it opt-in or opt-out for the ConPTY consumer. Per https://github.com/microsoft/terminal/issues/293#issuecomment-458239831, this is one of those sad compatibility hacks.

Because legacy console applications could never emit colors that weren't in the default 16, we chose to make their explicit requests for backgrounds that matched the "default" buffer background (in the legacy 4-bit sense) and foreground into requests for the "default" when they came out of the PTY. 😕

Reply:
If there weren't so many apps with a dependency on the layout of a console text attribute or the `SetConsoleTextAttributes` API, I'd be gung-ho for it. After all, conhost itself has a mode that lets us differentiate them. It does look rather garish, however:

(from Rich's blog post about that)
![image](https://user-images.githubusercontent.com/14316954/80160546-c5438280-8582-11ea-8bd7-13f72fae502c.png)